demo/style_configurator.c

```diff
@@ -0,0 +1,815 @@
+	}
+}
+
+// style_general? pass array in instead of static?
+static void
+style_global_colors(struct nk_context* ctx, struct nk_color color_table[NK_COLOR_COUNT])
+{
+	const char* color_labels[NK_COLOR_COUNT] =
+	{
+		"COLOR_TEXT:",
+		"COLOR_WINDOW:",
+		"COLOR_HEADER:",
+		"COLOR_BORDER:",
+		"COLOR_BUTTON:",
+		"COLOR_BUTTON_HOVER:",
+		"COLOR_BUTTON_ACTIVE:",
+		"COLOR_TOGGLE:",
+		"COLOR_TOGGLE_HOVER:",
+		"COLOR_TOGGLE_CURSOR:",
+		"COLOR_SELECT:",
+		"COLOR_SELECT_ACTIVE:",
+		"COLOR_SLIDER:",
+		"COLOR_SLIDER_CURSOR:",
+		"COLOR_SLIDER_CURSOR_HOVER:",
+		"COLOR_SLIDER_CURSOR_ACTIVE:",
+		"COLOR_PROPERTY:",
+		"COLOR_EDIT:",
+		"COLOR_EDIT_CURSOR:",
+		"COLOR_COMBO:",
+		"COLOR_CHART:",
+		"COLOR_CHART_COLOR:",
+		"COLOR_CHART_COLOR_HIGHLIGHT:",
+		"COLOR_SCROLLBAR:",
+		"COLOR_SCROLLBAR_CURSOR:",
+		"COLOR_SCROLLBAR_CURSOR_HOVER:",
+		"COLOR_SCROLLBAR_CURSOR_ACTIVE:",
+		"COLOR_TAB_HEADER:"
+	};
+
+	int clicked = 0;
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+	for (int i=0; i<NK_COLOR_COUNT; ++i) {
+		clicked |= style_rgb(ctx, color_labels[i], &color_table[i]);
+	}
+
+	if (clicked) {
+		nk_style_from_table(ctx, color_table);
+	}
+}
+
+static void
+style_text(struct nk_context* ctx, struct nk_style_text* out_style)
+{
+	struct nk_style_text text = *out_style;
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+	style_rgb(ctx, "Color:", &text.color);
+
+	style_vec2(ctx, "Padding:", &text.padding);
+
+	*out_style = text;
+}
+
+static void
+style_button(struct nk_context* ctx, struct nk_style_button* out_style, struct nk_style_button** duplicate_styles, int n_dups)
+{
+	struct nk_style_button button = *out_style;
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+	style_item_color(ctx, "Normal:", &button.normal);
+	style_item_color(ctx, "Hover:", &button.hover);
+	style_item_color(ctx, "Active:", &button.active);
+
+	style_rgb(ctx, "Border:", &button.border_color);
+	style_rgb(ctx, "Text Background:", &button.text_background);
+	style_rgb(ctx, "Text Normal:", &button.text_normal);
+	style_rgb(ctx, "Text Hover:", &button.text_hover);
+	style_rgb(ctx, "Text Active:", &button.text_active);
+
+	style_vec2(ctx, "Padding:", &button.padding);
+	style_vec2(ctx, "Image Padding:", &button.image_padding);
+	style_vec2(ctx, "Touch Padding:", &button.touch_padding);
+
+	const char* alignments[] =
+{
+	"LEFT",
+	"CENTERED",
+	"RIGHT",
+	"TOP LEFT",
+	"TOP CENTERED",
+	"TOP_RIGHT",
+	"BOTTOM LEFT",
+	"BOTTOM CENTERED",
+	"BOTTOM RIGHT"
+};
+
+#define TOP_LEFT       NK_TEXT_ALIGN_TOP|NK_TEXT_ALIGN_LEFT
+#define TOP_CENTER     NK_TEXT_ALIGN_TOP|NK_TEXT_ALIGN_CENTERED
+#define TOP_RIGHT      NK_TEXT_ALIGN_TOP|NK_TEXT_ALIGN_RIGHT
+#define BOTTOM_LEFT    NK_TEXT_ALIGN_BOTTOM|NK_TEXT_ALIGN_LEFT
+#define BOTTOM_CENTER  NK_TEXT_ALIGN_BOTTOM|NK_TEXT_ALIGN_CENTERED
+#define BOTTOM_RIGHT   NK_TEXT_ALIGN_BOTTOM|NK_TEXT_ALIGN_RIGHT
+
+	int aligns[] =
+{
+	NK_TEXT_LEFT,
+	NK_TEXT_CENTERED,
+	NK_TEXT_RIGHT,
+	TOP_LEFT,
+	TOP_CENTER,
+	TOP_RIGHT,
+	BOTTOM_LEFT,
+	BOTTOM_CENTER,
+	BOTTOM_RIGHT
+};
```

Comment:
This is probably a better way to write this :)
```c
struct {
    const char *name;
    int align;
} alignments = {
    { "LEFT", NK_TEXT_LEFT },
    ...
};


```

Reply:
Is it worth having to create a callback and use nk_combo_callback?  It would be nice if there were something between nk_combo and nk_combo_callback, something that you just give an offset and stride to so nuklear can access the strings directly.

Reply:
Yeah, I finally looked into this some more and I have a version stashed but I really think it's worse.  It's at least as long if not longer, more complicated, and has more disparate/dislocated parts.